File: android/src/main/java/com/getcapacitor/community/classes/events/EndpointEvent.java
package com.getcapacitor.community.classes.events;

import static com.getcapacitor.community.NearbyHelper.EndpointID;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import com.getcapacitor.community.classes.Endpoint;

public class EndpointEvent {

    @NonNull
    Endpoint endpoint;

    public EndpointEvent(@NonNull Endpoint endpoint) {
        this.endpoint = endpoint;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("endpointID", endpoint.endpointID());

        //        if (endpoint.endpointName() != null) {
        //            result.put("endpointName", endpoint.endpointName());
        //        }

        return result;
    }
}

File: ios/Sources/NearbyPlugin/Classes/Events/EndpointEvent.swift
import Foundation
import Capacitor

@objc public class EndpointEvent: NSObject {
    let endpoint: Endpoint

    init(_ endpoint: Endpoint) {
        self.endpoint = endpoint
    }

    public func toJSObject() -> JSObject {
        var result = JSObject()

        result["endpointID"] = endpoint.endpointID

        if let endpointName = endpoint.endpointName {
            if !endpointName.isEmpty {
                result["endpointName"] = endpointName
            }
        }

        if let endpointInfo = endpoint.endpointInfo {
            if !endpointInfo.isEmpty {
                result["endpointInfo"] = endpointInfo.base64EncodedString()
            }
        }

        if let rssi = endpoint.rssi {
            result["rssi"] = rssi
        }
        if let power = endpoint.power {
            result["power"] = power
        }
        if let distance = endpoint.distance {
            result["distance"] = distance
        }

        return result
    }
}
